plugins/html/icon_html/actions/markdown/action.py

```diff
@@ -2,30 +2,30 @@
 import pypandoc
 import base64
 import re
-from .schema import MarkdownInput, MarkdownOutput
+from .schema import MarkdownInput, MarkdownOutput, Input, Output, Component
 from insightconnect_plugin_runtime.exceptions import PluginException
 
 
 class Markdown(insightconnect_plugin_runtime.Action):
     def __init__(self):
         super(self.__class__, self).__init__(
             name="markdown",
-            description="Convert HTML to Markdown",
+            description=Component.DESCRIPTION,
             input=MarkdownInput(),
             output=MarkdownOutput(),
         )
 
     def run(self, params={}):
         tag_parser = "(?i)<\/?\w+((\s+\w+(\s*=\s*(?:\".*?\"|'.*?'|[^'\">\s]+))?)+\s*|\s*)\/?>"  # noqa: W605
-        tags = re.findall(tag_parser, params.get("doc"))
-        try:
-            if not len(tags):
-                raise PluginException(cause="Run: Invalid input.", assistance="Input must be of type HTML.")
-            output = pypandoc.convert_text(params.get("doc"), "md", format="html")
-            f = base64.b64encode(output.encode("ascii")).decode()
-            return {"markdown_contents": output, "markdown_file": f}
-        except Exception:
-            return {"error": "Error occurred please try again"}
+        doc = params.get(Input.DOC)
+        tags = re.findall(tag_parser, doc)
+
+        if not tags:
+            raise PluginException(cause="Invalid input.", assistance="Input must be of type HTML.")
 
-    def test(self):
-        return {"test": "test Success"}
+        try:
+            output = pypandoc.convert_text(doc, "md", format="html")
+        except RuntimeError as error:
+            raise PluginException(cause="Error converting doc file. ", assistance="Check stack trace log.", data=error)
+        f = base64.b64encode(output.encode("ascii")).decode()
+        return {Output.MARKDOWN_CONTENTS: output, Output.MARKDOWN_FILE: f}
```

Comment:
Avoid use of single letter variables, it dosent affect functionality but its good practice 😄 
```suggestion
        file_ = base64.b64encode(output.encode("ascii")).decode()
        return {Output.MARKDOWN_CONTENTS: output, Output.MARKDOWN_FILE: file_}
```